### ISSUE TITLE
Command not found when using Shortcut

### ISSUE THREAD

**Issue by Glujaz** [holzschu/a-shell#519]
Hi there,

the title says everything. I'm trying to run a script from shortcuts. But it seems that the commands can't be found when using the shortcut actions without opening a-Shell.

i try to us it for yt-dlp, but jump does not work as well.

any idea why ?

thanks !

**Comment by holzschu** [comment]
Is that all the commands (including `ls` or `pwd`), or just some commands?

`yt-dlp`, for example, is a Python script, so it is installed in a directory that is only visible from the app, not from the in-extension shortcut. `jump` is not installed for in-extension shortcuts (because it depends on bookmarks, and only the app has the ability to store bookmarks, not the in-extension shortcut). 

Keep in mind that in-extension shortcuts have a time limit of 5 seconds (which is very short, almost any python script goes over that, and `yt-dlp` definitely does). The best way to run long shortcuts, or shortcuts that depend on things that are available only from the app is to run shortcuts "in app" instead of "in extension", then call the command `open shortcuts://` which will go back to the Shortcuts app to execute the rest of the shortcut.

**Comment by holzschu** [comment]
Yes, these are iOS limitations. The 5 seconds rule is the most important one. The other is related to the fact that the extension and the app are two different apps, with different home directories and permissions. There are workarounds: you can copy Python packages from `~/Library/lib/python3.11/site-packages` in the app to `~shortcuts/Library/lib/python3.11/site-packages` (that's the extension home directory).

On the other question, I have no idea, but you might have more success by asking in the Discord server, there are some real experts of Shortcuts over there. The invitation is in the "help" command.
